### DISCUSSION TITLE
Regridding packages - what to use?

### DISCUSSION THREAD

**Discussion by kobebryant432** [CORDEX-be2/ValEnsPy#33] [Ideas]
Which packaging methodology to use for regridding.
Options that include conservative regridding (which are crucial!):
- CDO
- NCO
- [xESMF ](https://xesmf.readthedocs.io/en/stable/)(python wrapper for ESMF)

The disadvantage of the aforementioned packages are that they add significant, non-python dependencies. 
e.g. xESMF is not dask compatible!

I found this discussion on pangeo.
https://discourse.pangeo.io/t/conservative-region-aggregation-with-xarray-geopandas-and-sparse/2715

and I found this package (python only) that does this for rectilinear grids: https://github.com/EXCITED-CO2/xarray-regrid/tree/main

Will try this first one for now.

**Comment by kobebryant432** [comment]
https://github.com/EXCITED-CO2/xarray-regrid/tree/main requires python 3.10 or higher

**Reply by kobebryant432** [reply]
See https://github.com/EXCITED-CO2/xarray-regrid/issues/37

**Comment by kobebryant432** [comment]
Another option is to use existing xarray methods for all regridding options and require a cdo dependency for conservative regriddings.

**Comment by Ivanderkelen** [comment]
This sounds most reasonable to me. To my experience, regridding can have impacts on the results. Ine the remapcon2 function in cdo, we have confidence. When using conservative remapping of CDO, we can check whether the other regridding methods give the same results (eg a nearest neigbour remapping in xarray vs cdo). 

- [ ] check regridding methods cdo vs xarray after implementation? (nice to have)
